### PR TITLE
feat: float mobile editor

### PR DESCRIPTION
**Pull request by f97**
Related Issues : #193

Screen Shot 2022-09-20 at 13 57 37

### CODE REVIEW

**Comment by boojack** [comment]
How about adding the responsible style directly for mobile instead of a new setting field?

**Comment by f97** [comment]
> How about adding the responsible style directly for mobile instead of a new setting field?

Yep, It's responsive, it's just that I think some people prefer mobile with default design over float design like this.

File: api/user_setting.go [hunk 1]
```diff
@@ -100,6 +105,23 @@ func (upsert UserSettingUpsert) Validate() error {
 		if invalid {
 			return fmt.Errorf("invalid user setting editor font style value")
 		}
+	} else if upsert.Key == UserSettingMobileEditorStyleKey {
+		mobileEditorStyleValue := "normal"
+		err := json.Unmarshal([]byte(upsert.Value), &mobileEditorStyleValue)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal user setting editor font style")
```
Comment (boojack) [inline api/user_setting.go:112]:
```suggestion
			return fmt.Errorf("failed to unmarshal user setting mobile editor style")
```

File: api/user_setting.go [hunk 2]
```diff
@@ -33,6 +37,7 @@ var (
 	UserSettingLocaleValue          = []string{"en", "zh", "vi"}
 	UserSettingMemoVisibilityValue  = []Visibility{Privite, Protected, Public}
 	UserSettingEditorFontStyleValue = []string{"normal", "mono"}
+	UserSettingMobileEditorValue    = []string{"normal", "float"}
```
Comment (boojack) [inline api/user_setting.go:40]: It should be `UserSettingMobileEditorStyleValue`.

**Review by boojack — CHANGES_REQUESTED** [review]

**Review by boojack — APPROVED** [review]
LGTM

**Pull request merged**